Automattic/newspack-plugin
# feat: update Stripe Settings dashboard UI

### All Submissions:






### Changes proposed in this Pull Request:



Updates our Stripe Settings dashboard page with a new "Payment Methods" page which lays the groundwork for showing UI for all payment gateways that Newspack fully supports (currently only Stripe and WooPayments). This PR only implements UI for Stripe, for now.

The new UI removes settings fields which are duplicative of settings in the Stripe payment gateway's own settings pages, in favor of a link out to those settings pages for configuration. It shows a "Connection status" indicator showing whether the payment gateway is not connected, partially connected (with API keys but not with the new OAuth method), or fully connected.

It also reorganizes the "Transaction fees" settings fields slightly, for better visual clarity since the UI is now more hierarchical than before.

### How to test the changes in this Pull Request:

1. Disable the Stripe gateway plugin if it's already installed/activated on your site.
2. Check out this branch/rebuild
3. In the admin, visit **Newspack > Reader Revenue**
4. Confirm that the "Stripe Settings" tab is now called "Payment Methods"
5. Visit "Payment Methods" and confirm you see a plugin installer (since the plugin is deactivated):

<img width="1349" alt="Screenshot 2024-10-18 at 3 32 50 PM" src="https://github.com/user-attachments/assets/40be4d49-1d6f-4bff-82c6-ec45fb1cdbd2">

6. Click "Activate" and confirm that the plugin installer component is replaced by a "Stripe" settings component. If you previously had Stripe connected, it will show as enabled already and with Connection Status and settings in the content area, otherwise it will just show a toggle switch to enable the payment gateway.

7. If your site was already connected to Stripe via API keys (previously the only option) but not OAuth, confirm that the Stripe component shows "Needs attention" under "Connection Status", along with our transaction fee settings:

<img width="1348" alt="Screenshot 2024-10-18 at 5 03 27 PM" src="https://github.com/user-attachments/assets/3b6f5c1e-6071-4900-8ab9-e89e95afc0ec">

Note: in this state (connected via API keys but not OAuth), the Stripe gateway will continue to work on the front-end for checkout, but this is not guaranteed to continue working beyond Oct. 29.

8. Delete your Stripe keys (via WP CLI: `wp option delete woocommerce_stripe_settings`). Refresh the page and confirm that "Connection Status" shows "Not connected" and that the transaction fee fields are no longer shown.

<img width="1347" alt="Screenshot 2024-10-18 at 4 52 42 PM" src="https://github.com/user-attachments/assets/c41f35f0-d383-49f8-ac4e-154c277a3d8c">

9. Click "Connect" and confirm that you're redirected to the Stripe payment gateway's settings page.
10. On this page you should see a dialogue to authenticate or reauthenticate with Stripe (the messaging may differ depending on whether you're already connected via API keys):

<img width="1071" alt="Screenshot 2024-10-21 at 9 29 46 AM" src="https://github.com/user-attachments/assets/057cf9a2-a64a-40ed-842c-3564399e3ff5">

10. Complete the new authentication flow (if connecting to a test Stripe account, you should be able to click a "skip this flow" button to avoid having to enter real banking info during the flow).
11. After you're redirected back to the Stripe gateway settings page, return to our **Newspack > Reader Revenue > Payment Methods** page. Confirm that the Connection Status dialogue now shows that you're connected (in test mode) and that the transaction fees fields are once again available:

<img width="1348" alt="Screenshot 2024-10-18 at 5 04 18 PM" src="https://github.com/user-attachments/assets/bba81603-9997-4992-9e4a-58b4aacbc38e">

11. Smoke test the "Collect transaction fees" toggle and the related transaction fee fields to ensure they persist after saving and change the modal checkout UI for donations as expected. Note that in this PR, toggling OFF the "Collect transaction fees" toggle will hide the other settings fields.

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208504842518107
